src/persist-client/src/fetch.rs

```diff
@@ -259,16 +240,33 @@ pub(crate) async fn fetch_batch_part_blob(
         blob.get(&blob_key).await
     })
     .instrument(get_span.clone())
-    .await
-    .ok_or(blob_key)?;
+    .await;
+    let value = value.unwrap_or_else(|| {
```

Comment:
Ah right, compaction is why we didn't have this here. I suppose we can still make compaction work because it fetches things in tasks, which catches the panic and turns it into an `Err`, but that doesn't feel great...